new python function `using_remote_theme()` needs an argument

The function introduced in #571 to skip checks when using remote theme needs to pass `source_dir` as an argument otherwise it fails:

```
Run make lesson-check-all
Traceback (most recent call last):
  File "/home/runner/work/lesson-example/lesson-example/bin/lesson_check.py", line 577, in <module>
    main()
  File "/home/runner/work/lesson-example/lesson-example/bin/lesson_check.py", line 118, in main
    if not using_remote_theme():
  File "/home/runner/work/lesson-example/lesson-example/bin/lesson_check.py", line 174, in using_remote_theme
    config_file = os.path.join(source_dir, '_config.yml')
NameError: name 'source_dir' is not defined
make: *** [Makefile:142: lesson-check-all] Error 1
``` 


https://github.com/carpentries/styles/blob/4df505249a27191c3475102cde4c150b0a131102/bin/lesson_check.py#L173-L176

--- a/bin/lesson_check.py
+++ b/bin/lesson_check.py
@@ -115,7 +115,7 @@ def main():
     check_source_rmd(args.reporter, args.source_dir, args.parser)
 
     args.references = {}
-    if not using_remote_theme():
+    if not using_remote_theme(args.source_dir):
         args.references = read_references(args.reporter, args.reference_path)
 
     docs = read_all_markdown(args.source_dir, args.parser)
@@ -170,7 +170,7 @@ def parse_args():
 
     return args
 
-def using_remote_theme():
+def using_remote_theme(source_dir):
     config_file = os.path.join(source_dir, '_config.yml')
     config = load_yaml(config_file)
     return 'remote_theme' in config

